amundsen_application/static/js/ducks/search/sagas.ts

```diff
@@ -39,9 +42,58 @@ import {
   updateFromInlineResult,
   setPageIndex, setResource,
 } from './reducer';
+import {
+  clearAllFilters,
+  setSearchInputByResource,
+  UpdateSearchFilter
+} from './filters/reducer';
 import { autoSelectResource, getPageIndex, getSearchState } from './utils';
 import { BrowserHistory, updateSearchUrl } from 'utils/navigationUtils';
 
+/**
+ * Listens to actions triggers by user updates to the filter state.
+ * For better user experience debounce the start of the worker as multiple updates can happen in < 1 second.
+ */
+export function* filterWatcher(): SagaIterator {
+  yield debounce(750, [UpdateSearchFilter.CLEAR_CATEGORY, UpdateSearchFilter.UPDATE_CATEGORY], filterWorker);
+};
+
+/*
+ * Executes a search on the current resource.
+ * Actions that trigger this worker will have updated the filter reducer.
+ * The updated filter state is applied in searchResourceWorker().
+ * Updates the search url to reflect the change in filters.
+ */
+export function* filterWorker(): SagaIterator {
+  const state = yield select(getSearchState);
+  const { search_term, selectedTab, filters } = state;
+  const pageIndex = getPageIndex(state)
+  yield put(searchResource(search_term, selectedTab, pageIndex));
+  updateSearchUrl({ filters, resource: selectedTab, term: search_term, index: pageIndex }, true);
+};
+
+/**
+ * Listens to actions triggers by application updates to both the filter state and search term.
+ * This is intended to be temporary code. searchResource saga restructring will allow us to consolidate this support.
+ */
+export function* filterWatcher2(): SagaIterator {
+  yield takeLatest(UpdateSearchFilter.SET_BY_RESOURCE, filterWorker2);
```

Comment:
Can we give this a better name?

*Just an idea*: What if we had one `UpdateSearch` action that could accept any subset of search fields (term, index, resource, filter). Then it would have logic to determine which API it needs to call, or what pieces of state to update.

From the component side, we could just call `UpdateSearch` with the relevant search field.

```
SearchBar submitted -> UpdateSearch({ term: "new value" })

Table Tag clicked -> UpdateSearch({ filter: { tag: "test" }, term: "", resource: "table" })

Pagination clicked -> UpdateSearch({ page_index: 3 })
```

For this approach, the filters would need a centralized place to gather the filter values and generate some JSON. When you update the state of an `InputFilter` or `CheckboxFilter`, it would need to tell the `SearchFilter` object to generate the filter JSON by reaching out to each `FilterSection` and ask for their JSON value. Then it could stitch together the full filter value and submit a `UpdateSearch({ filter: newFilters })`. 

Just a thought that occurred...

Reply:
Copying suggestion into the description for the follow up ticket.